### PR TITLE
Rename existing data-related utils / resources to avoid conflicts with upcoming MLflow Tracking data integration

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Rename existing data-related utils / resources to avoid conflicts with upcoming MLflow Tracking data integration

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [X] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/recipes/test_ingest_step.py [hunk 1]
```diff
@@ -205,8 +205,8 @@ def test_ingests_remote_http_datasets_with_multiple_files_successfully(tmp_path)
                         "skip_data_profiling": True,
                         "using": "csv",
                         "location": [
-                            "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/data/winequality-red.csv",
-                            "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/data/winequality-white.csv",
+                            "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-red.csv",
+                            "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-white.csv",
```
Comment (dbczumar) [inline tests/recipes/test_ingest_step.py:209]: This test won't pass until this PR is merged

File: tests/tensorflow/save_tf_estimator_model.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def save_tf_iris_model(tmpdir):
 
 
 def save_tf_categorical_model(tmpdir):
-    path = os.path.join(args.mlflow_repo_path, "tests/data/uci-autos-imports-85.data")
+    path = os.path.join(args.mlflow_repo_path, "tests/datasets/uci-autos-imports-85.data")
```
Comment (dbczumar) [inline tests/tensorflow/save_tf_estimator_model.py:132]: This test won't pass until this PR is merged

**Review by prithvikannan — APPROVED** [review]
LGTM. What is the process for making sure those tests post-merge?

**Comment by dbczumar** [comment]
> LGTM. What is the process for making sure those tests post-merge?

Great question! I'll kick off the corresponding GitHub action to make sure it passes

**Pull request merged**